### DISCUSSION TITLE
Foundry Fundamentals: Section 1 - 27. Zksync Deployment Execution Errors (local & zksync sepolia)

### DISCUSSION THREAD

**Discussion by MajorDomDePIN** [Cyfrin/foundry-full-course-cu#2228] [Q&A]
Hello everyone, 

I am running the course in a wsl 2, but I have saved the project folder in the windows partition and not in the WSL itself. 

Now I wanted to continue today at this point of the course and for this reason start the docker node for the second time and interact with it. The first time this worked without any problems. Now I get the following output depending on which private key of zksync I use:

command: 
forge create ./src/SimpleStorage.sol:SimpleStorage --private-key 0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356 --rpc-url http://127.0.0.1:8011 --legacy --zksync
output:
server returned an error response: error code 3: Execution error: Transaction HALT: Account validation error: Not enough balance for fee + value

command:
forge create ./src/SimpleStorage.sol:SimpleStorage --rpc-url http://127.0.0.1:8011 --private-key 0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 --legacy --zksync
output:
server returned an error response: error code 3: Execution error: Transaction HALT: Account validation error: Account validation returned invalid magic value. Most often this means that the signature is incorrect

im working with foundryup-zksync and just dont know why this issue happens, if someone can help me with this i would be more then happy

**Comment by EngrPips** [comment]
I am unsure what the problem is here, but let's look forward to @PatrickAlphaC suggestions as soon as they are available here.

**Reply by MajorDomDePIN** [reply]
Did you were able to rebuild the issue?

**Reply by EngrPips** [reply]
Works fine for me 

<img width="1440" alt="Screenshot 2024-08-12 at 15 23 38" src="https://github.com/user-attachments/assets/965ad775-7eff-433f-8539-61c760112fa6">

**Comment by MajorDomDePIN** [comment]
I was now able to execute the Forge Create command, for which it was necessary to store the appropriate sender for the private key in the foundry.toml file. 

Here I get the error directly, because there is no file in the broadcast folder for this transaction, unlike @PatrickAlphaC demonstrates in the following video.

**Comment by cromewar** [comment]
Hello @MajorDomDePIN 

That is an interesting error. Can you please delete the project cache folder and compile the project again? 

Also, be sure the place you are storing the private key is on the WSL machine, as you mentioned that this happens depending on where you stored the private key, I guess you added it first when you were running the project under the Windows folder structure and not on the WSL one, so can we.

1. Be sure the project is on the `/home` folder of your WSL,
2. Delete the cache folder and recompile.
3. remove the private key from the anvil and add it again using the WSL terminal. Be sure to use the same terminal because if you're using a docker instance, it might not count as the same; therefore, storing it in different locations and not working.
4. Try again and tell us if it worked.

If not, don't forget to post an update so we can try any other alternative.

**Comment by MajorDomDePIN** [comment]
Thank you for your Answer!

I recloned the project folder into my wsl so i only working inside of it and not in windows anymore, but this couldn't solve the problem. 

i tried to run the container for the zksync-docker-machine in wsl as well as in windows, but both times i had the same issues with both of them.

currently i only can deploy with the first private key which is listed on https://docs.zksync.io/build/test-and-debug/in-memory-node, if i try the second one instead i got a "not enought balance" error...
![grafik](https://github.com/user-attachments/assets/51c55ac4-5d4b-4d3a-9663-a41868247fbd)

Unfortunately, these private keys do not match the private keys that are listed in the docker container. 
![grafik](https://github.com/user-attachments/assets/b4dc4ec5-3553-422a-abd8-209972928e40)

Regarding the keys from the container, they seem to work completely arbitrarily or randomly

![grafik](https://github.com/user-attachments/assets/743058fc-2124-4974-bb45-0171d8779914)
![grafik](https://github.com/user-attachments/assets/bdf7630c-7d4c-479a-a824-50ddb1b0f97b)

here is a script i made to try to deploy to every private key from the domain as well as the local node:
https://github.com/MajorDomDePIN/Deployment_Test

And if a transaction goes through, no broadcast file is created for chain 260, as already mentioned


Besides this i also ran into the following issue when it comes to deploy on zksync sepolia:
here it showes that i have 0.4 eth on zksync, but when i try to deploy on zksync i get an "insuffidient funds error" in which it says my balance would be zero:

ubuntu@DESKTOP~/Cyrin_Kurs_WSL/Cyfrin_Foundry_Course/foundry-f23/foundry-simple-storage-f23$ cast balance 0x63e42Efbc0EAB83cFFc1A6fB3f6953F1C221EcA1 --rpc-url https://sepolia.era.zksync.dev
401784623175245868

ubuntu@DESKTOP:~/Cyrin_Kurs_WSL/Cyfrin_Foundry_Course/foundry-f23/foundry-simple-storage-f23$ forge create src/SimpleStorage.sol:SimpleStorage --account metamask --rpc-url https://sepolia.era.zksync.dev --chain 300 --legacy --zksync
[⠃] Using zksolc-1.5.1
[⠊] Compiling (zksync)
No files changed, compilation skipped
Enter keystore password:
Enter keystore password:
Error: 
server returned an error response: error code 3: insufficient funds for gas + value. balance: 0, fee: 26651616082267374, value: 0

**Reply by cromewar** [reply]
Can you print `account mata mask' and see if it's using the correct one with the correct signer?
Of course, I don't share it here, but can you verify?

This is a really strange error, especially the one making the Docker container work/not work randomly. 

Let me do some research on this.

**Reply by cromewar** [reply]
Hello @MajorDomDePIN, any update here?

**Reply by cromewar** [reply]
Closing this issue due to lack of follow-up.
